File: backend/api/schemas/documents.py
from datetime import datetime
from pydantic import BaseModel
from backend.domain.enums import DocumentStatus, DocumentType


class DocumentResponse(BaseModel):
    document_id: str
    session_id: str
    filename: str
    document_type: DocumentType
    content_type: str
    size_bytes: int
    storage_reference: str
    status: DocumentStatus
    created_at: datetime
    updated_at: datetime


class DocumentListResponse(BaseModel):
    documents: list[DocumentResponse]


class DocumentExtractionResponse(BaseModel):
    extraction_id: str
    document_id: str
    status: DocumentStatus
    extracted_text: str | None = None
    structured_data: dict[str, object] | None = None
    processed_at: datetime | None = None
    created_at: datetime
    updated_at: datetime

File: patient-web/src/api/documents.ts
import { apiRequest } from "./client";

export type DocumentType =
  | "IMAGING_REPORT"
  | "LAB_REPORT"
  | "MEDICAL_RECORD"
  | "OTHER"
  | "PRESCRIPTION";

export type DocumentStatus = "FAILED" | "PROCESSED" | "PROCESSING" | "UPLOADED";

export interface PatientDocument {
  document_id: string;
  session_id: string;
  filename: string;
  document_type: DocumentType;
  content_type: string;
  size_bytes: number;
  storage_reference: string;
  status: DocumentStatus;
  created_at: string;
  updated_at: string;
}

interface DocumentResponse {
  data: PatientDocument;
}

export async function uploadDocument(
  sessionId: string,
  file: File,
  documentType: DocumentType = "OTHER",
): Promise<PatientDocument> {
  const formData = new FormData();
  formData.append("file", file);

  const response = await apiRequest<DocumentResponse>(
    `/sessions/${encodeURIComponent(sessionId)}/documents?document_type=${encodeURIComponent(documentType)}`,
    {
      method: "POST",
      body: formData,
    },
  );

  return response.data;
}
